tests/rptest/tests/controller_log_limiting_test.py

```diff
@@ -0,0 +1,394 @@
+            if "OK" in out:
+                success_amount += 1
+        assert quota_error_amount == 0
+        assert success_amount > 0
+
+
+class ControllerPartitionMovementLimitTest(PartitionMovementMixin,
+                                           EndToEndTest):
+    def __init__(self, ctx, *args, **kwargs):
+        super(ControllerPartitionMovementLimitTest, self).__init__(
+            ctx,
+            *args,
+            extra_rp_conf={
+                "rps_limit_move_operations": 25,
+                "enable_controller_log_rate_limiting": True,
+                "controller_log_accummulation_rps_capacity_move_operations": 0,
+            },
+            **kwargs)
+        self._ctx = ctx
+
+    def peform_move(self, topic, partition):
+        old_assignments, new_assignment = self._dispatch_random_partition_move(
+            topic=topic.name, partition=partition)
+        return old_assignments == new_assignment
+
+    @cluster(num_nodes=3)
+    def test_move_partition_limit(self):
+        self.start_redpanda(num_nodes=3)
+
+        topic = TopicSpec(partition_count=3)
+        self.client().create_topic(topic)
+        try:
+            while (self.peform_move(topic, 0)):
+                pass
+        except HTTPError as err:
+            assert err.response.status_code == 429
+        else:
+            assert 0, "Too Many Requests error must be raised"
+
+
+class ControllerAclsAndUsersLimitTest(RedpandaTest):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args,
+                         extra_rp_conf={
+                             "rps_limit_acls_and_users_operations":
+                             OPERATIONS_LIMIT,
+                             "enable_controller_log_rate_limiting": True
+                         },
+                         **kwargs)
+
+    @cluster(num_nodes=3)
+    def test_create_user_limit(self):
+        rpk = RpkTool(self.redpanda)
+        for i in range(OPERATIONS_LIMIT * 2):
+            try:
+                rpk.sasl_create_user(
+                    f"testuser_{i}", "password",
+                    self.redpanda.SUPERUSER_CREDENTIALS.algorithm)
+            except RpkException as err:
+                if i >= OPERATIONS_LIMIT:
+                    assert 'Too many requests' in err.stderr
+                else:
+                    raise err
+            else:
+                if i >= OPERATIONS_LIMIT:
+                    assert 0, "Too Many Requests error must be raised"
+
+    @cluster(num_nodes=3)
+    def test_create_acl_limit(self):
+        client = KafkaCliTools(self.redpanda)
+        self.client().alter_broker_config(
+            {"rps_limit_acls_and_users_operations": 1}, incremental=True)
+        for i in range(3):
+            try:
+                client.create_cluster_acls("User" + str(i), "describe")
+            except HTTPError as err:
+                if i >= 1:
+                    assert err.response.status_code == 429
```

Comment:
can we just wait for first failed request and assert that it failed after `cnt > OPERATIONS_LIMIT` times ? I think now the test is going to be flaky

Reply:
Changed test to 0 limit because acl creation requests is too slow, the limit has time to update between requests